app/lib/actions/pulp3/capsule_content/remove_unneeded_repos.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Pulp3
+    module CapsuleContent
+      class RemoveUnneededRepos < Pulp3::AbstractAsyncTask
+        def plan(smart_proxy)
+          plan_self(:smart_proxy_id => smart_proxy.id)
+        end
+
+        def invoke_external_task
+          smart_proxy = SmartProxy.unscoped.find(input[:smart_proxy_id])
```

Comment:
Pulp3::Abstract provides a smart_proxy method that i think you can just call instead of doing your own lookup